dockerunit-core/src/main/java/com/github/qzagarese/dockerunit/DockerUnitRule.java

```diff
@@ -0,0 +1,135 @@
+package com.github.qzagarese.dockerunit;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+import com.github.qzagarese.dockerunit.discovery.DiscoveryProvider;
+import com.github.qzagarese.dockerunit.discovery.DiscoveryProviderFactory;
+import com.github.qzagarese.dockerunit.internal.UsageDescriptor;
+import com.github.qzagarese.dockerunit.internal.ServiceContextBuilder;
+import com.github.qzagarese.dockerunit.internal.ServiceContextBuilderFactory;
+import com.github.qzagarese.dockerunit.internal.reflect.UsageDescriptorBuilder;
+import com.github.qzagarese.dockerunit.internal.reflect.DependencyDescriptorBuilderFactory;
+import com.github.qzagarese.dockerunit.internal.service.DefaultServiceContext;
+
+public class DockerUnitRule implements TestRule {
+
+    private final Logger logger = Logger.getLogger(this.getClass().getSimpleName());
+
+    private static final Map<String, ServiceContext> activeContexts = new HashMap<>();
+
+    private final UsageDescriptorBuilder descriptorBuilder = DependencyDescriptorBuilderFactory.create();
+    private final ServiceContextBuilder contextBuilder = ServiceContextBuilderFactory.create();
+    private final DiscoveryProvider discoveryProvider;
+    private final String serviceContextName;
+    private ServiceContext discoveryContext;
+
+    
+    public static ServiceContext getServiceContext(String serviceContextName) {
+        return activeContexts.get(serviceContextName);
+    }
+    
+    public static ServiceContext getDefaultServiceContext() {
+        return activeContexts.values().stream().findAny().orElse(null);
```

Comment:
This is to allow parallel suites to run concurrently inside the JVM, which I don't really foresee for now. The active contexts are mapped based on the constructor parameter, however this map will usually have only one instance, because of the single instance of DockerUnitRule being used in most scenarios.

Indeed there should never be less than one instance, so this should throw an Exception instead.
It needs a proper Javadoc.